src/rpc/reflect/util.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019-2024 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+use std::fmt::Display;
+
+use serde::{de::Visitor, forward_to_deserialize_any, Deserialize, Deserializer};
+
+/// "Introspection" by tracing a [`Deserialize`] to see if it's [`Option`]-like.
+///
+/// This allows us to smoothly operate between rust functions that take an optional
```

Comment:
nit: I'm unsure that `operate between` is a thing, if I'm understanding the intention correctly `switch between` would be one way to put it.